### PR TITLE
Add specs for coverage clipboard class

### PR DESCRIPTION
**Pull request by KarinaMendez2**
#49

### CODE REVIEW

File: spec/clipboard_spec.rb [hunk 1]
```diff
@@ -1,11 +1,85 @@
...
+    it 'handles clipboard not empty and no block given' do
+      element_1.cut(to: my_clipboard)
+      expect{my_clipboard.each}.to raise_error(NoMethodError)
+    end
+  end
+
+  context '#sort_by' do
+    it 'returns empty object when no block given' do
+      element_1.cut(to: my_clipboard)
+      expect(my_clipboard.sort_by!).to be {}
```
Comment (jpslav) [inline spec/clipboard_spec.rb:75]: Ditto here on `{}` being read as a block being passed to the `be` method, not being read as a hash.

[`Array#sort_by!`](https://apidock.com/ruby/v2_5_5/Array/sort_by%21) returns an Enumerator is no block is given so I think our test should look for the same

```ruby
expect(my_clipboard.sort_by!).to be_a Enumerator
```

`sort_by!`

File: spec/clipboard_spec.rb [hunk 2]
```diff
@@ -1,11 +1,85 @@
...
+    end
+
+    it 'returns sorted object when block given' do
+      element_1.search('span').cut(to: my_clipboard)
+      expect(
+        my_clipboard.sort_by!{|element| element.text}.to_s
+      ).to eq("[<span>ABC</span>, <span>B</span>, <span>Element 1</span>, <span>Zebra</span>]")
```
Comment (jpslav) [inline spec/clipboard_spec.rb:82]: You can use `fake_element` here again and sort by the value of the paste method:  `my_clipboard.sort_by!(&:paste)`

**Review by jpslav — CHANGES_REQUESTED** [review]
Looking good!

File: spec/clipboard_spec.rb [hunk 3]
```diff
@@ -1,85 +1,76 @@
...
 
-    it 'handles clipboard not empty and no block given' do
-      element_1.cut(to: my_clipboard)
-      expect{my_clipboard.each}.to raise_error(NoMethodError)
+      it 'when clipboard is not empty and no block is given' do
+        my_clipboard.add(2)
+        expect(my_clipboard.each).to be my_clipboard
+      end
     end
   end
```
Comment (jpslav) [inline spec/clipboard_spec.rb:59]: Need a spec to make sure that each works.  This is a tricky one so I will share the answer and other devs should take a look too.

One way is more manual.  You can create a lambda which is a callable bit of code (here there is actually no code in the `{}` block).  And then you can expect the block to be called with the items on the clipboard.  And then you add the items and call each, passing in the block (`&a_block` does that).

```ruby
it 'iterates on elements ' do
  a_block = ->(item) {}
  expect(a_block).to receive(:call).with("howdy")
  expect(a_block).to receive(:call).with("y'all")
  my_clipboard.add("howdy").add("y'all").each(&a_block)
end
```

It turns out that [rspec has a built-in way to test blocks](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/yield-matchers#yield-with-args-matcher) with less code:

```ruby
it 'iterates on elements' do
  my_clipboard.add("howdy").add("y'all")
  expect{ |block| my_clipboard.each(&block) }.to yield_successive_args("howdy", "y'all")
end
```

File: spec/clipboard_spec.rb [hunk 4]
```diff
@@ -1,85 +1,76 @@
...
 
   context '#sort_by' do
     it 'returns empty object when no block given' do
-      element_1.cut(to: my_clipboard)
-      expect(my_clipboard.sort_by!).to be {}
+      my_clipboard.add(fake_element(2))
```
Comment (jpslav) [inline spec/clipboard_spec.rb:63]: don't think you need this line

File: spec/clipboard_spec.rb [hunk 5]
```diff
@@ -1,85 +1,76 @@
...
 
     it 'returns sorted object when block given' do
-      element_1.search('span').cut(to: my_clipboard)
+      my_clipboard.add(fake_element('Zebra'))
+      my_clipboard.add(fake_element('ABC'))
+      my_clipboard.add(fake_element('Element 1'))
       expect(
-        my_clipboard.sort_by!{|element| element.text}.to_s
-      ).to eq("[<span>ABC</span>, <span>B</span>, <span>Element 1</span>, <span>Zebra</span>]")
+        my_clipboard.sort_by!(&:paste)[0].paste
```
Comment (jpslav) [inline spec/clipboard_spec.rb:72]: `Clipboard` has a `paste` method itself (as do the fake elements), so if you take out the `[0]` you can test that the other elements are sorted too `'ABCElement 1Zebra'`.

**Review by jpslav — CHANGES_REQUESTED** [review]

File: spec/clipboard_spec.rb (hunk 5 above)
Comment (KarinaMendez2) [reply spec/clipboard_spec.rb:72]: Here I was actually getting `undefined method `paste' for #<Array:'...'>` so I printed the result to the console, it comes out like: `#<OpenStruct paste="ABC"> #<OpenStruct paste="Element">#<OpenStruct paste="Zebra">`

**Review by jpslav — APPROVED** [review]

**Pull request merged**